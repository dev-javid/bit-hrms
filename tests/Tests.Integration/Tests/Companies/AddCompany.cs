namespace Tests.Integration.Tests.Companies
{
    public class AddCompany(TestWebApplicationFactory<Program> factory) : IntegrationTest(factory)
    {
        private const string Route = "/api/companies";

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public async Task Add_Company()
        {
            await LoginAsSuperAdminAsync();

            var command = new
            {
                Name = "Company Name",
                AdministratorName = "Administrator Name",
                Address = "Some Random Place on the Planet Earth",
                Email = "company@example.com",
                PhoneNumber = "0123456789",
            };

            HttpResponseMessage response;
            using (StringContent content = new StringContent(JsonSerializer.Serialize(command), Encoding.UTF8, JsoneMediaType))
            {
                response = await Client.PostAsync(Route, content);
            }

            response.StatusCode.Should().Be(HttpStatusCode.OK);
            var result = await response.Content.ReadAsStringAsync();
            Approvals.VerifyJson(result.Ignore("$.companyId"));
        }
    }
}
