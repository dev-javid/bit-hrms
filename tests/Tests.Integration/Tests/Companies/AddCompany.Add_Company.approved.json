{
    "weeklyOffDays": [],
    "companyId": "__IGNORED_VALUE__",
    "name": "Company Name",
    "phoneNumber": "0123456789",
    "financialMonth": 1,
    "administratorName": "Administrator Name",
    "email": "company@example.com",
    "address": "Some Random Place on the Planet Earth"
}